src/components/User-Tables/index.tsx

```diff
@@ -56,16 +57,26 @@ export default function UserTables({ users, isEditor }: propType) {
               <StyledTableCell align="center">{user.email}</StyledTableCell>
               <StyledTableCell align="center">{user.role}</StyledTableCell>
               <StyledTableCell align="center">{user.job}</StyledTableCell>
-              {isEditor && (
-                <StyledTableCell
-                  align="center"
-                  onClick={() =>
-                    navigate('/edit-user', { state: { userId: user.id } })
-                  }>
-                  <EditButton data-testid="edit-user-button" />
-                  Editar
-                </StyledTableCell>
-              )}
+ 
+                {isAdmin &&
+                  (<StyledTableCell
+                    align="center"
+                    onClick={() =>
+                      navigate('/edit-user', { state: { userId: user.id } })
+                    }>    
+                    <EditButton data-testid="edit-user-button" />
+                    <>Editar</>
+                  </StyledTableCell>)}
+
+                {!isAdmin && isEditor && user.role !== 'administrador' &&
```

Comment:
Existe alguma diferença entre essas duas renderizações? Se não, não dá para simplificar em somente uma condição e diminuir a duplicação de código?

Reply:
Não tem diferença na renderização em si, mas tem diferença em quando ela aparece pra alguns e para outros não. No caso, é para aparecer o botão de editar em todos os usuários, se o usuário autenticado for o admin. Caso o usuário autenticado seja o gerente, o botão de editar aparece em todos os usuários, exceto nos que são administradores

Tentei mexer de algumas formas diferentes, mas infelizmente foi essa a que deu certo. Se alguém souber melhorar, pode mudar.

Reply:
Descobri que utilizando o parêntesis funciona sem erros. Alterei nos dois arquivos